### ISSUE TITLE
Using minRange with stepsArray

### ISSUE THREAD

**Issue by Stalyon** [angular-slider/ng5-slider#60]
I cannot use property "minRange" (or "maxRange") when I use property "stepsArray".
Example here: https://stackblitz.com/edit/ng5-slider-simple-slider-example-sbx52j

Error in console: 
> ERROR TypeError: Cannot read property 'value' of undefined
>     at SliderComponent.getStepValue (slider.component.ts:592)
>     at SliderComponent.applyHighValue (slider.component.ts:616)
>     at SliderComponent.positionTrackingHandle (slider.component.ts:2233)
>     at SliderComponent.onMove (slider.component.ts:1888)
>     at ehMove (slider.component.ts:1809)
>     at HTMLDocument.eval (dom_renderer.ts:52)
>     at ZoneDelegate.invokeTask (zone.js:421)
>     at Object.onInvokeTask (ng_zone.ts:262)
>     at ZoneDelegate.invokeTask (zone.js:420)
>     at Zone.runTask (zone.js:188)

**Comment by piotrdz** [comment]
Thanks for spotting this. I see that when using `stepsArray` , `minRange` and `maxRange` are treated as index offsets in the steps array instead of being treated as values. I'll fix that and look for other places where the same bug may be present.

**Comment by Stalyon** [comment]
Thanks for your great work.
Waitting for your fix and the release.

**Comment by Stalyon** [comment]
Hi.
Do you have a date about a fix and a release?

**Comment by piotrdz** [comment]
Sorry, it's going to take a few more days. I'm having a really busy time.

**Comment by piotrdz** [comment]
I pushed a change that fixes this problem with `minRange`/`maxRange`, but I still want to fix other instances of similar bugs: when using `minLimit`/`maxLimit` and when using `pushRange`. I should have it ready in a couple of days.

**Comment by piotrdz** [comment]
I a sorry for the long delay. I have been incredibly busy recently, and only got back to this now.

Now that I looked again at the code, I have decided that this was not the right way to go with this.  I have reverted my previous change and I intend to keep the code as it was - with `minRange` / `maxRange` being indexes into `stepsArray`.

If you use `stepsArray`, the code is already doing exactly that - `floor`, `ceil` and `step` are internally set to be index-based (`floor=0`, `ceil=stepsArray.length()-1`, `step=1`). Other variables used throughout the code are also meant to follow suit. The value of each step only comes into play when translating the values to and from the `value` and `highValue` bindings.

To add code that handles this differently would make the slider positioning code quite complicated, and difficult to reason about. There is a high chance that some more use-cases with other options will slip through and invite more bugs.

So instead, I would advise to use index offsets to solve the particular problem in your example ([StackBlitz](https://stackblitz.com/edit/ng5-slider-issue-60-workaround?file=src/app/app.component.ts)).

I have updated the documentation to make it clearer what options are supported with stepsArray and how they should be specified (as values or indexes into `stepsArray`).
